From ab905c8d5b8694acd25587ec142d3d74ef6c483c Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Tue, 23 Sep 2025 04:59:32 -0400
Subject: [PATCH] Adjust high DPI scaling handling for Qt 6 to avoid
 deprecation warning (#116)

---
 rendercanvas/qt.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/rendercanvas/qt.py b/rendercanvas/qt.py
index 4f9cd4b..b915108 100644
--- a/rendercanvas/qt.py
+++ b/rendercanvas/qt.py
@@ -159,7 +159,11 @@ def enable_hidpi():
         except Exception:
             pass  # fail on non-windows
     try:
-        QtWidgets.QApplication.setAttribute(QtCore.Qt.AA_EnableHighDpiScaling, True)
+        # https://doc.qt.io/qtforpython-6/faq/porting_from2.html#class-function-deprecations
+        # > High DPI is by default enabled in Qt 6 and cannot be turned off.
+        dpi_scaling_not_deprecated = qt_version_info[0] < 6
+        if dpi_scaling_not_deprecated:
+            QtWidgets.QApplication.setAttribute(QtCore.Qt.AA_EnableHighDpiScaling, True)
     except Exception:
         pass  # fail on older Qt's
 
